src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/resource/SmokeTestResource.kt

```diff
@@ -64,8 +64,11 @@ class SmokeTestResource(
     @PathVariable(value = "testProfile")
     testProfile: String,
   ): Flux<TestStatus> = runCatching { PsiTestProfiles.valueOf(testProfile).profile }
-    .map { profile -> psiSmokeTestService.runSmokeTest(profile).doOnComplete { psiSmokeTestService.cleanup(profile).share().block() } }
-    .getOrDefault(Flux.just(TestStatus("Unknown test profile $testProfile", FAIL)))
+    .map { profile ->
+      psiSmokeTestService.runSmokeTest(profile).concatMap {
+        psiSmokeTestService.cleanup(profile).thenReturn(it)
```

Comment:
The `onComplete` was doing so without a publisher so was blocking the thread indefinitely.